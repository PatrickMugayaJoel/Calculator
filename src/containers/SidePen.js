import React, { Component } from 'react';
import { toast } from 'react-toastify';
import Form from "../components/Form";
import axios from 'axios';

const MakePost = postData => {
    return axios.post("http://api.mathjs.org/v4/", postData)
        .then(resp => {
          toast.error("Success");
        })
        .catch(e => {
          toast.error("Error!");
        });
}

export class SidePen extends Component {
    constructor (props) {
      super(props);
      this.state = {
        num_one: "",
        num_two: "",
        operation: "add"
      };
    }

    onChange = event => {
        this.setState({ [event.target.name]: event.target.value });
    }

    onSubmit = e => {
        e.preventDefault();

        let operator = "";
        switch (this.state.operation) {
          case "add":
            operator = "+";
            break;
          case "sub":
            operator = "-";
            break;
          case "mul":
            operator = "*";  
            break;        
          case "div":
            operator = "/";
            break;
          default:
              operator = "+";
        }

        const postData = { 
            expr: `${this.state.num_one} ${operator} ${this.state.num_two}`, 
            precision: 14 
        };

        MakePost(postData);
    }

    render () {
        const props = {
            onSubmit: this.onSubmit,
            onChange: this.onChange,
            state: this.state
        };
        return (
          <div className="col-md-4">
            <div className="side-pannel full-height">
              <span className="title">Numbers</span>
              <div className="form">
                <Form props={props} />
              </div>
            </div>
          </div>
        );
    }
}

export default SidePen;