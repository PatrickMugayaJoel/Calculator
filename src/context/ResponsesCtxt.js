import React, {useState, createContext} from 'react';

export const ResponseContext = createContext();

export const ResponseProvider = props => {
    const [responces, setResponces] = useState([]);
    return (
        <ResponseContext.Provider value={[responces, setResponces]}>
            {props.children}
        </ResponseContext.Provider>
    );
}
