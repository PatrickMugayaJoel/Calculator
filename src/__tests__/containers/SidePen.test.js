import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import SidePen from '../../containers/SidePen';
import {ResponseProvider} from "../../context/ResponsesCtxt";

const wrapper = mount(
    <ResponseProvider>
      <SidePen />
    </ResponseProvider>
  );

describe('SidePen container snapshot', () => {
    it('matches snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe('Rendering form', () => {
    it('Should render form', () => {
        const form = wrapper.find('Form');
        expect(form).toHaveLength(1);
    });
});
