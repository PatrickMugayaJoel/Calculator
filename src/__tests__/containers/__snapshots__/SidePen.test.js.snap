// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidePen container snapshot matches snapshot 1`] = `
<ResponseProvider>
  <SidePen>
    <div
      className="col-sm-4"
    >
      <div
        className="side-pannel full-height"
      >
        <span
          className="title"
        >
          Numbers
        </span>
        <div
          className="form"
        >
          <Form
            props={
              Object {
                "onChange": [Function],
                "onSubmit": [Function],
                "state": Object {
                  "num_one": "",
                  "num_two": "",
                  "operation": "add",
                },
              }
            }
          >
            <form
              onSubmit={[Function]}
            >
              <label>
                Number one
              </label>
              <input
                className="input"
                name="num_one"
                onChange={[Function]}
                placeholder="Enter number one"
                required={true}
                type="number"
                value=""
              />
              <label>
                Number two
              </label>
              <input
                className="input"
                name="num_two"
                onChange={[Function]}
                placeholder="Enter number two"
                required={true}
                type="number"
                value=""
              />
              <label>
                Operation
              </label>
              <select
                className="input"
                name="operation"
                onChange={[Function]}
                type="text"
              >
                <option
                  value="add"
                >
                  ADD
                </option>
                <option
                  value="sub"
                >
                  SUB
                </option>
                <option
                  value="mul"
                >
                  MUL
                </option>
                <option
                  value="div"
                >
                  DIV
                </option>
              </select>
              <input
                className="submit"
                type="submit"
                value="Post"
              />
            </form>
          </Form>
        </div>
      </div>
    </div>
  </SidePen>
</ResponseProvider>
`;
